glambie/util/timeseries_helpers.py

```diff
@@ -593,3 +596,115 @@ def get_average_trends_over_new_time_periods(start_dates, end_dates, changes, ne
+    date_gaps = [
+        datetime.datetime.strptime(interpolated_dataframe['end_date'][i], '%d/%m/%Y') - datetime.datetime.strptime(
+            interpolated_dataframe['start_date'][i], '%d/%m/%Y') for i in range(len(interpolated_dataframe))]
+    date_gaps_in_days = [a.days for a in date_gaps]
+    interpolated_dataframe['days_covered'] = date_gaps_in_days
+
+    # calculate change per day in each row
+    interpolated_dataframe['glacier_change_per_day'] = [
+        a / b for a, b in zip(interpolated_dataframe.glacier_change_observed, interpolated_dataframe.days_covered)]
+    interpolated_dataframe['glacier_change_uncertainty_per_day'] = [
+        a / b for a, b in zip(interpolated_dataframe.glacier_change_uncertainty, interpolated_dataframe.days_covered)]
+
+    dates_list_fractional = datetime_dates_to_fractional_years(new_start_dates)
+    interpolated_dataframe['date_fractional'] = dates_list_fractional
+
+    # Linear interpolation of glacier_change_per_day to fill gaps
+    df = pd.DataFrame({'time': interpolated_dataframe['date_fractional'],
+                       'mass': interpolated_dataframe['glacier_change_per_day'],
+                       'error': interpolated_dataframe['glacier_change_uncertainty_per_day'],
+                       'hydro': interpolated_dataframe['hydrological_correction_value'],
+                       'sea_level': interpolated_dataframe['sea_level_correction_value']})
+    df_int = df.interpolate(method='linear')
+
+    interpolated_dataframe['glacier_change_per_day'] = df_int['mass'].values.tolist()
+    interpolated_dataframe['glacier_change_uncertainty_per_day'] = df_int['error'].values.tolist()
+    interpolated_dataframe['hydrological_correction_value'] = df_int['hydro']
+    interpolated_dataframe['sea_level_correction_value'] = df_int['sea_level']
+
+    # Convert back to glacier_change_observed
+    interpolated_dataframe['glacier_change_observed'] = [
+        a * b for a, b in zip(interpolated_dataframe.glacier_change_per_day, interpolated_dataframe.days_covered)]
+    interpolated_dataframe['glacier_change_uncertainty'] = [
+        a * b for a, b in zip(interpolated_dataframe.glacier_change_uncertainty_per_day,
+                              interpolated_dataframe.days_covered)]
+
+    # removing all remaining small day gaps by setting end_date(i) = start_date(i+1)
+    updated_end_dates = []
+    for i in range(len(interpolated_dataframe.end_date) - 1):
+        updated_end_dates.append(interpolated_dataframe.start_date[i + 1])
+    updated_end_dates.append(interpolated_dataframe['end_date'].tolist()[-1])
+    interpolated_dataframe['end_date'] = updated_end_dates
+
+    # then remove row that has been interpolated for GRACE gap, as this value will be much bigger than the rest
+    # (check for start date == 1/7/2017, not super robust)
+    interpolated_dataframe.drop(
+        interpolated_dataframe.loc[interpolated_dataframe.start_date.__eq__('01/07/2017')].index, inplace=True)
```

Comment:
Would be good to add a bit more context here, why specifically are you removing this date? has it caused issues in the past?

Reply:
Have added some explanation - this approach will likely change in the future, but for now the simplest thing to do is just remove this row!